WordPress/src/main/java/org/wordpress/android/ui/uploads/PostUploadHandler.java

```diff
@@ -279,12 +266,26 @@ protected Boolean doInBackground(PostModel... posts) {
 
             RemotePostPayload payload = new RemotePostPayload(mPost, mSite);
 
-            if (sShouldRemoteAutoSavePosts.contains(mPost.getId())) {
-                mDispatcher.dispatch(PostActionBuilder.newRemoteAutoSavePostAction(payload));
-            } else {
-                mDispatcher.dispatch(PostActionBuilder.newPushPostAction(payload));
+            switch (UploadUtils.getPostUploadAction(mPost)) {
```

Comment:
DO_NOTHING doesn't exist in this branch yet. I'll fix this issue in one of the following PRs.

Reply:
Btw it's quite interesting as Lint should be failing on this issue 🤷‍♂.

Reply:
Fixed in 98fcbd8cb2b453cc6ad1840dc148fb64de7206ad -> I decided to follow the fail fast principle and throw an exception since the `getPostUploadAction` doesn't even return DO_NOTHING.